tests/README.md

```diff
@@ -0,0 +1,92 @@
+# s2n-tls testing
+s2n-tls has a wide variety of tests covering a wide variety of different functionality. This document serves as a "quick reference", providing an overview of all the testing that s2n-tls does and linking to additional documentation.
+
+## Unit Tests
+_Is the internal functionality of s2n-tls correct?_
+
+These test the correctness of low-level s2n-tls functions. These functions are commonly private. We have unit tests for both the C library and the [rust bindings](../bindings/rust). 
+
+### Sanitizers
+The C unit tests are additionally run with the following sanitizers enabled
+- Valgrind: used to detect memory errors
+- Address Sanitizer (ASAN): used to detect memory errors
+- Thread Sanitizer (TSAN): used to detect race conditions
+
+Further information:
+- [C tests](unit/)
+- [Valgrind memcheck documentation](https://valgrind.org/docs/manual/mc-manual.html)
+- [ASAN documentation](https://clang.llvm.org/docs/AddressSanitizer.html)
+- [TSAN documentation](https://clang.llvm.org/docs/ThreadSanitizer.html)
+
+## Integration Tests
+_Is the functionality of s2n-tls correct and compatible with other implementations?_
+
+Our integration tests involve interoperation with other TLS implementations, because testing against another implementation improves confidence that s2n-tls has implemented the feature correctly and is compliant with relevant RFCs.
+
+s2n-tls integration tests are run using our integrationv2 framework. Tests are implemented using "client" and "server" executables from various implementations. The client and server interactions are coordinated with pytest.
+
```

Comment:
Nit: might be nice to list some examples so readers have something more concrete than "implementations".
```suggestion
s2n-tls integration tests are run using our integrationv2 framework. Tests are implemented using "client" and "server" executables from various implementations like Openssl and Boringssl. The client and server interactions are coordinated with pytest.

```